### ISSUE TITLE
LaTeX math migration guide

### ISSUE THREAD

**Issue by jason-s** [typst/typst#162]
Has anyone started a LaTeX math migration guide to help those of us familiar with LaTeX to typeset our math with typst instead? (e.g. showing LaTeX and typst equivalents)

I'd be willing to contribute.

**Comment by SeniorMars** [comment]
This is a great idea. If someone could port this: https://joshua.smcvt.edu/undergradmath/, I feel like it would be a great start.

**Comment by jason-s** [comment]
or https://math.meta.stackexchange.com/questions/5020/mathjax-basic-tutorial-and-quick-reference

**Comment by jason-s** [comment]
I can start a gist on Github for a couple of them, and (again) would be willing to contribute here and there. (But don't have time to really polish it up or aim for any kind of completeness.)

**Comment by jason-s** [comment]
https://gist.github.com/jason-s/f3333130f1e38cf41e68ea82ccd45621

**Comment by johanvx** [comment]
> This is a great idea. If someone could port this: https://joshua.smcvt.edu/undergradmath/, I feel like it would be a great start.

I've just made a Typst port at https://github.com/johanvx/typst-undergradmath. Have a check if this interests you.

**Comment by SeniorMars** [comment]
> > This is a great idea. If someone could port this: [joshua.smcvt.edu/undergradmath](https://joshua.smcvt.edu/undergradmath/), I feel like it would be a great start.
> 
> I've just made a Typst port at [johanvx/typst-undergradmath](https://github.com/johanvx/typst-undergradmath). Have a check if this interests you.

You should ask to link this as a PR.

**Comment by Enivex** [comment]
https://typst.app/docs/guides/guide-for-latex-users/

**Comment by jason-s** [comment]
Hey, wait a minute, that's not what I was looking for when I opened this issue.

**Comment by jason-s** [comment]
The key here is that new users need a migration guide specifically for the math part, not just LaTeX in general.

**Comment by Enivex** [comment]
> The key here is that new users need a migration guide specifically for the math part, not just LaTeX in general.

The guide does cover math. I'm not sure what you're looking for exactly.

**Comment by jason-s** [comment]
And one of the most important points is to highlight key differences, for example:

- `epsilon`/`epsilon.alt` and `phi`/`phi.alt` follow Unicode which is opposite of TeX's choices for epsilon/varepsilon and phi/varphi
- Typst `dot` vs TeX `cdot`, and dots in general
- Typst `macron` vs TeX `bar`
- Typst `integral.cont` vs TeX `oint`

(etc.)

**Comment by Enivex** [comment]
> And one of the most important points is to highlight key differences, for example:
> 
> - `epsilon`/`epsilon.alt` and `phi`/`phi.alt` follow Unicode which is opposite of TeX's choices for epsilon/varepsilon and phi/varphi
> - Typst `dot` vs TeX `cdot`, and dots in general
> - Typst `macron` vs TeX `bar`
> - Typst `integral.cont` vs TeX `oint`
> 
> (etc.)

I'd argue that the important part is highlighting big picture syntax differences between latex and typst, not minute details in the naming of symbols. Those can be looked up on the symbols page (which is mentioned), or using suggestions and autocompletion.

**Comment by jason-s** [comment]
> The guide does cover math. I'm not sure what you're looking for exactly.

It does and it doesn't. The "How do I input maths?" section from https://typst.app/docs/guides/guide-for-latex-users/ is a very good introduction but it only covers a few important points.

Typst-undergradmath (https://github.com/johanvx/typst-undergradmath/releases/tag/v0.0.2) is a more in-depth intro for someone coming to Typst who hasn't used LaTeX before, but the big issue is pointing out the differences. There's a lot of similarities, and for someone coming from LaTeX, the simplest way to learn is a two-pronged approach, one highlighting the general forms, which I think you have, and the other highlighting the exceptions. (My German teacher used to say "Same as in English!" for words like Karotte and Apfel and Haus, and they are kind of the same, but slightly different, and then you have 'It's all Greek to me' vs. `Es kommt mir Spanisch vor`)

From my original post:

>Has anyone started a LaTeX math migration guide to help those of us familiar with LaTeX to typeset our math with typst instead? (e.g. showing LaTeX and typst equivalents)

I was hoping it would include a section like this with some examples highlighting some of the differences: https://gist.github.com/jason-s/f3333130f1e38cf41e68ea82ccd45621

(I'd be willing to contribute, but this spring has been hellishly busy for me at my job. If there's an overall document organization, I can contribute a few examples.)

**Comment by jason-s** [comment]
> I'd argue that the important part is highlighting big picture syntax differences between latex and typst, not minute details in the naming of symbols. Those can be looked up on the symbols page (which is mentioned), or using suggestions and autocompletion.

But that contributes to cognitive load; I'd have to look up the symbols page every time I want to learn something. I really just want to see the odd cases. Like I said, I think the simplest way for someone coming from LaTeX is to learn via a two-pronged approach, one highlighting the general forms (present already) and the other highlighting the exceptions.

Please don't just say "look at the symbols page to see what's the same/different".

**Comment by Enivex** [comment]
Since the guide has already been created, I think it would be best if you start a new issue with concrete suggested improvements to what you'd like to see changed in the current migration guide.

(My personal opinion is that I think an exhaustive list of differences between typst and latex is unnecessary. In fact I think it would contribute to cognitive overload, not alleviate it, but ultimately it's also not up to me.)

**Comment by johanvx** [comment]
I attempted to write a somewhat concrete LaTeX math migration guide, but soon found that it would easily become a list of differences between Typst and LaTeX, which was boring and annoying.

I do think that such a guide helps, but I'm afraid that it won't be just a section of the genereal guide. What I suggest is a deep discussion before starting the math guide.

**Comment by jason-s** [comment]
>I do think that such a guide helps, but I'm afraid that it won't be just a section of the genereal guide.

Agreed, it should be targeted for math, in a bit more depth.

> What I suggest is a deep discussion before starting the math guide.
 
Agreed.

> My personal opinion is that I think an exhaustive list of differences between typst and latex is unnecessary.

I'm not asking for exhaustive. Something that lets a LaTeX reader help understand and remember key differences to get them to handle 95 - 99% of situations. Right now it's maybe around 80%.
